nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -455,14 +514,14 @@ in
       { enable = cfg.buildMachines != [];
         text =
           concatMapStrings (machine:
-            "${if machine ? sshUser then "${machine.sshUser}@" else ""}${machine.hostName} "
-            + machine.system or (concatStringsSep "," machine.systems)
```

Comment:
I am very against using `builtins.currentSystem` here, not just because it is impure, but because we are describing *other* systems---what the current system is is by-definition irrelevant!

Reply:
Absolutely agreed! I don't know what I was thinking :) I'll fix it and bring this PR up to speed.

Reply:
Great! Excited to see this get merged soon :)